empire.go

```diff
@@ -602,8 +642,12 @@ func (opts DeployOpts) Validate(e *Empire) error {
 
 // Deploy deploys an image and streams the output to w.
 func (e *Empire) Deploy(ctx context.Context, opts DeployOpts) (*Release, error) {
+	if err := authorizeApp(ctx, "Deploy", opts.App); err != nil {
```

Comment:
This is currently exploitable, if you deploy without explicitly specifying an app:

If I had a policy like so:

```go
var policy = acl.Policy{
        acl.Statement{
                Effect: acl.Deny,
                Action: []string{"Deploy"},
                Resource: []string{"app-name"},
        },
}
```

Then if I knew the app that a Docker image would get resolved to, I would still be able to deploy.

I think this authorization needs to be moved post resolution.